bliss/simulator/simulated_dataset.py

```diff
@@ -87,3 +91,76 @@ def val_dataloader(self):
 
     def test_dataloader(self):
         return DataLoader(self, batch_size=None, num_workers=self.num_workers)
+
+
+class CachedSimulatedDataset(pl.LightningDataModule, IterableDataset):
+    def __init__(
+        self,
+        n_batches: int,
+        valid_n_batches: int,
+        fix_validation_set: bool,
+        num_workers: int,
+        batch_size: int,
+        file_data_capacity: int,
+        cached_data_path: str,
+    ):
+        super().__init__()
+
+        self.n_batches = n_batches
+        self.valid_n_batches = valid_n_batches
+        self.fix_validation_set = fix_validation_set
+        self.num_workers = num_workers
+        self.batch_size = batch_size
+        self.file_data_capacity = file_data_capacity
+        self.cached_data_path = cached_data_path
+
+        self.data: List[FileDatum] = []
+        self.valid: Union[List[FileDatum], CachedSimulatedDataset] = []
+
+        # assume cached image files exist, read from disk
+        for filename in os.listdir(self.cached_data_path):
+            if "valid" in filename:
+                continue
+            if filename.startswith("dataset") and filename.endswith(".pkl"):
+                self.data += self.read_file(f"{self.cached_data_path}/{filename}")
+        assert self.data, "No cached data loaded; run `generate.py` first"
+
+        if self.fix_validation_set:
```

Comment:
Probably better to assume that we're always using a fixed validation set, if we're using a fixed training set.

Reply:
Done.